### PR TITLE
feat(forms): add `@daffodil/forms` and `daffFormNotEmptyValidator`

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds a validator for checking if a form has any field filled out.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
This is useful for updates that operate on partials. Any single field can make the whole form valid but there must be at least one valid field.

### CODE REVIEW

**Comment by griest024** [comment]
put in new forms package

**Comment by griest024** [comment]
@damienwebdev moved to forms package and improved to handle form controls as well as groups

File: libs/forms/src/validators/not-empty.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+
+/**
+ * A validator for a form group that returns an error if all of its immediate children don't have a truthy value.
+ * Empty strings are considered falsy.
+ */
+// TODO: handle nested form groups
```
Comment (damienwebdev) [inline libs/forms/src/validators/not-empty.ts:12]: Is this Todo still correct?

File: libs/forms/src/validators/not-empty.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import {
+  AbstractControl,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+
+/**
+ * A validator for a form group that returns an error if all of its immediate children don't have a truthy value.
```
Comment (damienwebdev) [inline libs/forms/src/validators/not-empty.ts:9]: This doesn't return an error.

What do you think about:

"A form validator that passes validation when the Form control or at least one of the immediate children of a Form group have a value."

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/forms/src/validators/not-empty.ts (hunk 1 above)
Comment (griest024) [reply libs/forms/src/validators/not-empty.ts:12]: yeah I think ideally this validator would handle nested form groups. Do you think it should be implemented for this PR? I don't have an immediate use case for it.

File: libs/forms/src/validators/not-empty.ts (hunk 2 above)
Comment (griest024) [reply libs/forms/src/validators/not-empty.ts]: changed

**Pull request merged**